### PR TITLE
Virtual keyboard: add a setting to disable swipes

### PR DESCRIPTION
**Pull request by hius07**
Default to enable. Fixes https://github.com/koreader/koreader/issues/10976.

Cleanup.

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/11066)

### CODE REVIEW

File: frontend/ui/widget/virtualkeyboard.lua [hunk 1]
```diff
@@ -422,8 +422,9 @@ function VirtualKey:onHoldSelect()
 end
 
 function VirtualKey:onSwipeKey(arg, ges)
+    if G_reader_settings:isFalse("keyboard_swipes_enabled") then return true end
     Device:performHapticFeedback("KEYBOARD_TAP")
-    if self.flash_keyboard and not self.skipswipe then
```
Comment (Frenzie) [inline frontend/ui/widget/virtualkeyboard.lua:426]: Iirc that's intended for internal use (like on the shift key), regardless whether it's fully hooked up at this time. Basically see https://github.com/koreader/koreader/blob/58cb1ff42b522011b98c4bccd114ce11d24eaf9d/frontend/ui/widget/virtualkeyboard.lua#L62-L64

Comment (Frenzie) [reply frontend/ui/widget/virtualkeyboard.lua:426]: But removing it is probably fine given it hasn't served any purpose the past x years.

**Comment by poire-z** [comment]
The rest (95% :)) of changes in menu_keyboard_layout.lua is just refactoring, no UI/feature change, right?

**Comment by hius07** [comment]
Minor UI changes:
-"layout-specific settings" goes up into the "layout" group
-"Keyboard **appearance** settings" - they are about the keyboard exterior

The main refactoring is: to iterate over both an array and a hash-table, I've changed "FFIUtil.orderedPairs + util.arrayContains" to "ipairs + table.sort(array)".

**Review by poire-z — APPROVED** [review]
^ ok, not gonna follow all this, trusting you :)

**Comment by NiLuJe** [comment]
> "ipairs + table.sort(array)".

Haven't checked the code, but I assume you meant `table.sort(hash)`, otherwise I'm missing something obvious?

**Comment by hius07** [comment]
> Haven't checked the code, but I assume you meant `table.sort(hash)`, otherwise I'm missing something obvious?

Array (active layouts) is not sorted alphabetically.

**Pull request merged**